lib/ecto/association.ex

```diff
@@ -233,6 +233,18 @@ defmodule Ecto.Association do
               {[changeset|changesets], structs, halt, valid?}
             {:ok, struct} ->
               {[changeset|changesets], [struct | structs], halt, valid?}
+            {:error, {error_changeset, :insert_join}} ->
+              # Roll the errors originating from a join into the parent_changeset
+              case changesets do
```

Comment:
Shouldn't we add the errors to the `changeset` instead of revisiting the `changesets` list?